xds/src/main/java/io/grpc/xds/RouteMatch.java

```diff
@@ -48,6 +51,37 @@
         Collections.<HeaderMatcher>emptyList(), null);
   }
 
+  /**
+   * Returns {@code true} if a request with the given path and headers passes all the rules
+   * specified by this RouteMatch.
+   *
+   * <p>The request's headers are given as a key-values mapping, where multiple values can
+   * be mapped to the same key.
+   *
+   * <p>Match is not deterministic if a runtime fraction match rule presents in this RouteMatch.
+   */
+  boolean matches(String path, Map<String, Set<String>> headers) {
+    if (!pathMatch.matches(path)) {
+      return false;
+    }
+    for (HeaderMatcher headerMatcher : headerMatchers) {
+      if (!headers.containsKey(headerMatcher.getName())) {
+        return false;
```

Comment:
What if `headerMatcher.presentMatch.equals(Boolean.FALSE)`? Or what if `headerMatcher.presentMatch.equals(Boolean.TRUE)` and `headerMatcher.isInvertedMatch` is true?